### PR TITLE
JAMES-3486: MailboxChangesMethod test adaptation for distributed environment

### PR DESCRIPTION
**Pull request by LanKhuat**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621-integration-tests/memory-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/memory/MemoryMailboxChangesMethodTest.java [hunk 1]
```diff
@@ -26,10 +26,24 @@
 import org.apache.james.JamesServerExtension;
 import org.apache.james.jmap.api.change.MailboxChange;
 import org.apache.james.jmap.rfc8621.contract.MailboxChangesMethodContract;
+import org.apache.james.mailbox.model.MailboxId;
 import org.apache.james.modules.TestJMAPServerModule;
 import org.junit.jupiter.api.extension.RegisterExtension;
 
 public class MemoryMailboxChangesMethodTest implements MailboxChangesMethodContract {
+    static class TestId implements MailboxId {
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/memory-jmap-rfc-8621-integration-tests/src/test/java/org/apache/james/jmap/rfc8621/memory/MemoryMailboxChangesMethodTest.java]: Use MemoryId?

File: mailbox/api/src/main/java/org/apache/james/mailbox/RightManager.java [hunk 2]
```diff
@@ -85,6 +85,8 @@
 
     MailboxACL listRights(MailboxPath mailboxPath, MailboxSession session) throws MailboxException;
 
+    MailboxACL listRights(MailboxId mailboxId, MailboxSession session) throws MailboxException;
```
Comment (chibenwa) [inline mailbox/api/src/main/java/org/apache/james/mailbox/RightManager.java:88]: :+1: +1000 !!!! <3

File: server/container/guice/protocols/jmap/src/main/java/org/apache/james/jmap/draft/JmapGuiceProbe.java [hunk 3]
```diff
@@ -26,6 +26,7 @@
 import org.apache.james.core.Username;
```
Comment (chibenwa) [inline server/container/guice/protocols/jmap/src/main/java/org/apache/james/jmap/draft/JmapGuiceProbe.java:26]: ` JAMES-XXXX Adapt MailboxChangesMethodContract for stability against distributed environment`

I'm picky but can you edit the commit message to explain what adaptation you did? Why it is necessary? Why it do not lead to errors?

**Review by chibenwa — APPROVED** [review]

**Comment by LanKhuat** [comment]
`
Failed to execute goal on project event-sourcing-pojo: Could not resolve dependencies for project org.apache.james:event-sourcing-pojo:jar:3.6.0-SNAPSHOT: Failed to collect dependencies at com.google.guava:guava:jar:25.1-jre -> com.google.j2objc:j2objc-annotations:jar:1.1: Failed to read artifact descriptor for com.google.j2objc:j2objc-annotations:jar:1.1: Could not transfer artifact com.google.j2objc:j2objc-annotations:pom:1.1 from/to central (https://repo.maven.apache.org/maven2): /root/.m2/repository/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.pom.part (No such file or directory)
`

**Comment by LanKhuat** [comment]
test this please

File: server/data/data-jmap/src/test/java/org/apache/james/jmap/api/change/MailboxChangeRepositoryContract.java [hunk 4]
```diff
@@ -522,9 +570,10 @@ default void getChangesShouldReturnDelegatedChanges() {
 
         assertThat(repository.getSinceStateWithDelegation(ACCOUNT_ID, referenceState, Optional.empty()).block().getUpdated())
             .containsExactly(id1);
-
     }
 
+
+
```
Comment (Arsnael) [inline server/data/data-jmap/src/test/java/org/apache/james/jmap/api/change/MailboxChangeRepositoryContract.java:576]: extra lines

**Review by Arsnael — APPROVED** [review]

**Comment by LanKhuat** [comment]
`
org.jenkinsci.plugins.workflow.cps.steps.ParallelStepException: Parallel step 5821046ca50d6a1e7e8d4f597d3185fb7c82fff5 failed
	at org.jenkinsci.plugins.workflow.cps.steps.ParallelStep$ResultHandler$Callback.checkAllDone(ParallelStep.java:153)
	at org.jenkinsci.plugins.workflow.cps.steps.ParallelStep$ResultHandler$Callback.onFailure(ParallelStep.java:119)
	at org.jenkinsci.plugins.workflow.cps.CpsBodyExecution$FailureAdapter.receive(CpsBodyExecution.java:294)
	at com.cloudbees.groovy.cps.impl.ThrowBlock$1.receive(ThrowBlock.java:68)
	at com.cloudbees.groovy.cps.impl.ConstantBlock.eval(ConstantBlock.java:21)
`
test this please

**Comment by Arsnael** [comment]
Was green before CI restart

**Comment by Arsnael** [comment]
test this please

**Comment by LanKhuat** [comment]
`
[c5553e9770b0f123454e5c37f1f093f67eb30c9a] [ERROR] Errors: 
[c5553e9770b0f123454e5c37f1f093f67eb30c9a] [ERROR]   RabbitMQDeletedMessageVaultIntegrationTest>DeletedMessageVaultIntegrationTest.vaultEndpointShouldRestoreImapDeletedMailbox:255 » ConditionTimeout
`
test this please

**Comment by chibenwa** [comment]
See https://github.com/linagora/james-project/pull/4186

**Pull request closed without merging**